### PR TITLE
Implement Debug.Filter and Debug.Sink

### PR DESCRIPTION
**Pull request by FelonEkonom**
Related Jira ticket: https://membraneframework.atlassian.net/browse/MC-174

### CODE REVIEW

File: test/membrane/integration/simple_elements_test.exs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @impl true
+    def handle_parent_notification({:send_buffers, number}, _ctx, state) do
+      buffers =
+        Enum.map(1..number, fn i ->
+          %Buffer{payload: inspect(i)}
+        end)
+
+      {[buffer: {:output, buffers}], state}
+    end
+  end
```
Comment (mat-hek) [inline test/membrane/integration/simple_elements_test.exs:34]: Does this do anything beyond the testing source?

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane/simple/filter.ex [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  alias Membrane.StreamFormat
+
+  def_input_pad :input, accepted_format: _any, flow_control: :auto
+  def_output_pad :output, accepted_format: _any, flow_control: :auto
+
+  @spec identity(any()) :: any()
+  def identity(arg), do: arg
+
+  def_options handle_buffer: [spec: (Buffer.t() -> Buffer.t()), default: &__MODULE__.identity/1],
+              handle_event: [spec: (Event.t() -> Event.t()), default: &__MODULE__.identity/1],
```
[the remark this answers is not in the document]
Comment (mat-hek) [reply lib/membrane/simple/filter.ex:34]: BTW, you can use ` &Function.identity/1

File: test/membrane/integration/simple_elements_test.exs (hunk 1 above)
Comment (FelonEkonom) [reply test/membrane/integration/simple_elements_test.exs:34]: This source works in `push`, while Testing Source works in `pull`

Comment (mat-hek) [reply test/membrane/integration/simple_elements_test.exs:34]: does it matter in this use case?

Comment (mat-hek) [reply test/membrane/integration/simple_elements_test.exs:34]: It wouldn't be more than you provide via the `output` option

File: lib/membrane/lambda_filter.ex [hunk 3]
```diff
@@ -1,16 +1,11 @@
-defmodule Membrane.Simple.Filter do
+defmodule Membrane.LambdaFilter do
```
Comment (mat-hek) [inline lib/membrane/lambda_filter.ex:1]: ❤️

File: lib/membrane/lambda_filter.ex [hunk 4]
```diff
@@ -27,14 +22,30 @@ defmodule Membrane.Simple.Filter do
...
+                Function with arity 1, that maps buffers handled by this filter.
+                Result of this function is passed to the `:buffer` action on `:output` pad.
+                """
+              ],
+              handle_event: [
+                spec: (Event.t() -> Event.t() | [Event.t()]),
+                default: &Function.identity/1,
+                description: """
+                Function with arity 1, that maps events handled by this filter.
+                Result of this function is passed to the `:event` action on the opposite pad to
```
Comment (mat-hek) [inline lib/membrane/lambda_filter.ex:38]: hmm, this way we cannot know whether the event is going downstream or upstream, which is pretty important

File: lib/membrane/debug/filter.ex [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  @impl true
+  def handle_event(pad, event, _ctx, state) do
+    _ingored = state.handle_event.(event)
+
+    opposite_pad =
+      case pad do
+        :input -> :output
+        :output -> :input
+      end
```
Comment (mat-hek) [inline lib/membrane/debug/filter.ex:74]: you can use the forward action instead

File: lib/membrane/debug/filter.ex [hunk 6]
```diff
@@ -0,0 +1,84 @@
+defmodule Membrane.Debug.Filter do
+  @moduledoc """
+  Membrane Filter, that can be used to create a child that will be used to debug data flowing thouth pipeline.
```
Comment (mat-hek) [inline lib/membrane/debug/filter.ex:3]:
```suggestion
  Membrane Filter, that can be used to create a child that will be used to debug data flowing thouth pipeline.
  
```

**Review by mat-hek — APPROVED** [review]

**Pull request merged**